scanners/zap/zap.py

```diff
@@ -476,30 +476,33 @@ def _setup_spider(self):
     def _setup_ajax_spider(self):
         """Prepare an spiderAjax job and append it to the job list"""
 
-        if self.my_conf("spiderAjax", default=False) is False:
+        params = self.config.subtree_to_dict(self.absolute_conf_path("spiderAjax"))
+        if params is None:
             return
 
-        af_spider_ajax = {
+        job = {
             "name": "spiderAjax",
             "type": "spiderAjax",
-            "parameters": {
-                "user": Zap.USER if self.authenticated else "",
-                "maxDuration": self.my_conf("spiderAjax.maxDuration", default=0),
```

Comment:
With this change, the default values for `maxDuration` and `url` are not defined if these parameters are not set, unlike other settings in this block, such as `user` or `browserId`. Is this because they already have default values when not specified in the configuration ?

Reply:
Yes, maxDuration is optional in ZAP. In absent, ZAP makes it 0 (unlimited), so it does not change.
As for the `url`, it defaults to the first context's URL, which RapiDAST assigns the `application.url` value, and is mandatory in rapidast.